### ISSUE TITLE
Upgrade to the latest react-docgen

### ISSUE THREAD

**Issue by ryanswanson** [sapegin/react-styleguidist#180]
The latest version of react-docgen includes fixes for support of union types in JSDoc, which would be really great to have supported in react-styleguidist as well.

[https://github.com/reactjs/react-docgen/commits/v2.10.0](https://github.com/reactjs/react-docgen/commits/v2.10.0)

I discovered this by tracking down the following console error (styleguidist server console - not JavaScript console).

```bash
Error when parsing src/components/select/Select.js
TypeError: Cannot read property 'name' of undefined
```

Here is an example of the offending JSDoc comment: Note the union type of {(string|number)}.

```JavaScript
/**
 * Retrieve a value from the given options and valueKey
 * @param {(string|number)} value - the selected value(s)
 * @param {Object} props - the Select component's props (or nextProps)
 */
expandValue(value, props) {
  ...
}
```

**Comment by sapegin** [comment]
Pull request would make it faster.

**Comment by mik01aj** [comment]
btw. maybe an update would fix https://github.com/sapegin/react-styleguidist/issues/37 too.

**Comment by nhavar** [comment]
They just merged code for HOC support as well which might fix #143 (https://github.com/reactjs/react-docgen/pull/124)

**Comment by sapegin** [comment]
@nhavar Yep, we’ll see very son.

**Comment by sapegin** [comment]
Fixed in #189.
